### ISSUE TITLE
Design Intent - Accessibility Feedback - IIR, COE, v3 Form Update

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#108213]
## Next Steps for the VFS team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://ows.io/qs/gb1dfnkl) at any time._

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [x] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Thank you for the comprehensive presentation! Very helpful and easy to follow on the recording.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Should:** Avoid conditional display of information based on the radio button selected, which I think is what you had in mind on the [certificate use page](https://www.figma.com/design/I40qWfEd6F4CSPfDSMqGeQ/COE-2024-25-Updates?node-id=1632-127393&t=9ZiDJKzyL1fbRAhE-4). I think the intentions there make sense, but there's a good chance that screen reader users are going to miss those definitions entirely.
   
   Basically, if I'm a screen reader user moving through a long form flow, I'm likely going to try to move through it as efficiently as I can. I'm probably not going to have my screen reader announce the whole text of the page because that's going to be long and cumbersome, and on many pages won't give me any new information. Instead, when I land on a new page in the flow I'm going to hit [tab] to move through the interactive elements (inputs, links, buttons) and then move on to the next page, unless there's something that catches my attention to tell me I should look at something in more detail. But that means after I select "Cash-out refinance" or whatever, I'll press [tab] and jump straight to the finish later/back/continue buttons at the bottom, skipping over the definitions without ever realizing they're there.
   
   And even if I do realize that those definitions are there, I may not realize that they're changing based on the radio button I select, because there's nothing out of the box that will notify screen reader users that content has updated dynamically. You could code that using `aria-live`, but that's a pretty heavy solution when I think there are some lighter-touch options available.
   
   Use the [tile variant](https://design.va.gov/storybook/?path=/story/uswds-va-radio--tile) of the radio button, which lets you bundle description text in with each radio button. Users get the definitions they need in context with each option, and descriptions are programmatically associated with the radio buttons so screen reader users will hear them as they move through the options.
   
   You could also try putting those definitions in an accordion, which provides a tab stop for screen reader users and a gentle prompt to listen to the extra content that's in the accordion rather than just moving on. But again, I'd advise against displaying that content conditionally.
   
   If conditional display is really the best way to go, then let me know at your Midpoint Review and I can provide notes on what I'd look for with an `aria-live` region.
- [ ] **Consider:** There are a few places in the flow where you have helper text that appears above the form inputs, eg. the [name explanation on the first page](https://www.figma.com/design/I40qWfEd6F4CSPfDSMqGeQ/COE-2024-25-Updates?node-id=1611-21348&t=9ZiDJKzyL1fbRAhE-4) and the [property address on the loan details page](https://www.figma.com/design/I40qWfEd6F4CSPfDSMqGeQ/COE-2024-25-Updates?node-id=1632-127554&t=9ZiDJKzyL1fbRAhE-4). Similar concerns apply for this --- if it's not an interactive element, a screen reader user may miss it.
   
   That could be fine, or it could be a problem! I don't know your problem space well enough to say either way, so that gets to be your job to decide. The line I would typically draw is that if it's helper text that's critical to understanding and completing a task (or even just very likely that someone will get lost without that text), we definitely need to find a way to announce it to screen reader users. If it's redundant text or just a nice-to-have clarifying text, then it's not a problem if screen reader users are more likely to miss it.
   
   If you've got text that absolutely needs to be announced, then your best options are hint text on the question (usually the `hint` prop on a VA web component) or description text on a radio or checkbox tile. For helper text at the start of a page, there's also another option available: bundling the helper text inside a `legend` within the `fieldset` that contains the inputs. Anything inside the `legend` will be announced the first time you tab into the `fieldset`, so screen reader users should pretty reliably hear it. That would look something like:
   ```
   <fieldset>
      <legend>
         <h3 class="some-style">Heading text</h3>
         <p class="some-other-style">Helper text</p>
      </legend>
      [inputs]
   </fieldset>
   ```

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone
- [ ] Add the 'collaboration-cycle-mobile-app' label if this feedback is for the mobile app

**Comment by hubgitian** [comment]
**Should: Avoid conditional display of information based on the radio button selected, which I think is what you had in mind on the [certificate use page](https://www.figma.com/design/I40qWfEd6F4CSPfDSMqGeQ/COE-2024-25-Updates?node-id=1632-127393&t=9ZiDJKzyL1fbRAhE-4). I think the intentions there make sense, but there's a good chance that screen reader users are going to miss those definitions entirely.**

Conditional display of info fix:
- [x] https://github.com/department-of-veterans-affairs/va-iir/issues/1651

**Consider: There are a few places in the flow where you have helper text that appears above the form inputs, eg. the [name explanation on the first page](https://www.figma.com/design/I40qWfEd6F4CSPfDSMqGeQ/COE-2024-25-Updates?node-id=1611-21348&t=9ZiDJKzyL1fbRAhE-4) and the [property address on the loan details page](https://www.figma.com/design/I40qWfEd6F4CSPfDSMqGeQ/COE-2024-25-Updates?node-id=1632-127554&t=9ZiDJKzyL1fbRAhE-4). Similar concerns apply for this --- if it's not an interactive element, a screen reader user may miss it.**

- [x] name explanation fix:
https://github.com/department-of-veterans-affairs/va-iir/issues/1647

- [x] address explanation fix: The file has been annotated for the engineers to use the <fieldset> code

<img width="1239" alt="Image" src="https://github.com/user-attachments/assets/a30c9917-6f19-4284-b388-81f4d4931e84" />
